src/getting-started/using-esp-generate.md

```diff
@@ -0,0 +1,52 @@
+# Using `esp-generate`
+
+With all the necessary tools installed, you're ready to create your first Rust project for ESP32.
+
+## Generating a Project
+
+To start, launch the interactive configuration tool by running:
+
+```shell
+esp-generate --chip esp32c3 your-project-name
+```
+
+<!-- 0.4.0 doesn't include the full interactive mode, yet --->
+Make sure to replace the chip type and project name as needed. You can also leave them out and `esp-generate` will prompt you for them.
+
+<!---
+I think a screenshot would make all this look a bit more interesting - but the drawback is it will "always" be outdated.
+--->
+
+> 💡 **Tip**: Prefer a non-interactive setup? You can specify configuration options directly using the `-o/--options` flag.
+>
+> Additionally, using the `--headless` flag will prevent the TUI (Terminal User Interface) from launching:
+> ```shell
+> esp-generate --chip esp32 -o alloc -o wifi --headless your-project-name
+> ```
+
+Adjust the options as needed for your project. See [Available Options][available-options] section of the README.
+
+[available-options]: https://github.com/esp-rs/esp-generate?tab=readme-ov-file#available-options
+
+> 💡 **Hint**: When using `espflash` you might want to enable `Use esp-backtrace as the panic handler.` and `Use the log crate to print messages.` under `Flashing, logging and debugging (espflash)`
```

Comment:
Maybe we should only use tip or hint, I dont see much difference between them (?). We could also include it on https://github.com/esp-rs/book/blob/main/rust-doc-style-guide.md#admonitions